CRM/Report/Form/Instance.php

```diff
@@ -130,9 +130,14 @@ public static function buildForm(&$form) {
 
       // prepare user_roles to save as names not as ids
       if (function_exists('user_roles')) {
-        $user_roles_array = user_roles();
-        foreach ($user_roles_array as $key => $value) {
-          $user_roles[$value] = $value;
+        if (function_exists('user_role_names')) {
```

Comment:
Mind adding a comment explaining why? (Drupal8) This is piling on to code that is already difficult to read, and will need cleanup at some point.

Reply:
oops, I think I missed this notification. I've updated the PR with an explanation for d8. Sorry for the delay @mlutfy